### PR TITLE
Various improvements to the documentation.

### PR DESCRIPTION
**Pull request by csadorf**
- Extend documentation of H5Store and H5StoreManager.
 - Extend documentation of DictManager.
 - Expose JSONDict, H5Store, H5StoreManager in API.
 - Misc. fixes and improvements.

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -114,21 +113,40 @@ The Job class
...
 
+
+The H5Store
+===========
+
+This class implements the interface to the job's :attr:`~signac.contrib.job.Job.data` attribute, but can also be used stand-alone:
+
+.. autoclass:: H5Store
+    :members:
+
```
Comment (bdice) [inline doc/api.rst]: I'm not sure if this affects anything but it looks like there is one empty line here, compared to two empty lines before "The H5Store" and after "The H5StoreManager"

File: signac/core/h5store.py [hunk 2]
```diff
@@ -257,11 +257,25 @@ class H5Store(MutableMapping):
...
+
+        with H5Store('fileh5').open(mode='r') as h5s:
+            pass
+
     :param filename:
         The filename of the underlying HDF5-file.
     :param kwargs:
-        Additional keyword arguments to be forward to the h5py.File constructor
-        See documentation for :class:`h5py.File` for more information.
+        Additional keyword arguments to be forward to the ``h5py.File`` constructor
```
Comment (bdice) [inline signac/core/h5store.py]: forward ➡️ forwarded
Also add a period after "constructor."

File: signac/core/h5store.py [hunk 3]
```diff
@@ -257,11 +257,25 @@ class H5Store(MutableMapping):
...
+        See the documentation for the
+        `h5py.File constructor <http://docs.h5py.org/en/latest/high/file.html#File>`_
```
Comment (bdice) [inline signac/core/h5store.py]: Should this link use intersphinx instead?

File: signac/core/h5store.py [hunk 4]
```diff
@@ -342,13 +356,30 @@ def close(self):
 
     @property
     def file(self):
+        """Access the underlying instance of h5py.File.
+
+        This property exposes the underlying ``h5py.File`` object enabling
```
Comment (bdice) [inline signac/core/h5store.py]: Use intersphinx.

File: signac/core/h5store.py [hunk 5]
```diff
@@ -342,13 +356,30 @@ def close(self):
...
+        use of functions such as ``create_dataset()`` or ``requires_dataset()``.
+
+        .. note::
+
+            The store must be open to access this property!
+
+        :returns:
+            The ``h5py`` file-object that this store is operating on.
+        :rtype:
+            ``h5py.File``
```
Comment (bdice) [inline signac/core/h5store.py]: # inter-
image

File: doc/api.rst (hunk 1 above)
Comment (csadorf) [reply doc/api.rst]: It doesn't look like it, but either way it wouldn't matter.

File: signac/core/h5store.py (hunk 3 above)
Comment (csadorf) [reply signac/core/h5store.py]: I tried, doesn't look they expose it properly. We can possibly fix that in a subsequent release.

File: signac/core/h5store.py (hunk 4 above)
Comment (csadorf) [reply signac/core/h5store.py]: See comment above.

File: signac/core/h5store.py (hunk 5 above)
Comment (csadorf) [reply signac/core/h5store.py]: See comment above.

**Pull request merged**